src/libraries/System.Drawing.Primitives/src/System/Drawing/KnownColorTable.cs

```diff
@@ -495,14 +499,43 @@ public static uint GetSystemColorArgb(KnownColor color)
         {
             Debug.Assert(Color.IsKnownColorSystem(color));
 
-            return ColorTranslator.COLORREFToARGB(Interop.User32.GetSysColor((byte)ColorValueTable[(int)color]));
+            if (!SystemColors.s_useAlternativeColorSet || HighContrastEnabled())
+            {
+                return ColorTranslator.COLORREFToARGB(Interop.User32.GetSysColor((byte)ColorValueTable[(int)color]));
+            }
+
+            int index = color <= KnownColor.WindowText ? (int)color : (int)color - (int)KnownColor.ButtonFace + (int)KnownColor.WindowText + 1;
```

Comment:
Could you include a comment describing what you're doing here?  Also, consider a helper since I see this twice.